CMSgov/onemac
# OY2 22729: Inactivated RAI Responses should be fully ignored

Story: https://qmacbis.atlassian.net/browse/OY2-22729
Endpoint: https://d21sxccb1vnkpk.cloudfront.net
(not sure why the github comment didn't work on this PR)

### Details
When Inactivating some RAI Responses, we discovered that the details of the Inactivated RAI Response were getting pulled into the package.  This is because the return statement was blocked by the status check.

### Changes
- moved the status check to Before the timestamp check (the newness of an inactivated event is irrelevant)
- included the softDeleteComponent function because building this function is how we found the bug and we could not use the function for the PROD data changes, so makes sense to include it here.
- softDeleteComponent function inactivates a component and updates the package accordingly
- updated the package builder to delete a package item if it should no longer be a package on the dashboard

### Test Plan
1. Login as a submitting user
2. Navigate to Package View
3. Find or create some components to inactivate, both package and RAI Response
4. login to AWS console
5. Use the softDeleteComponent lambda to inactivate the components
6. Verify the package details now ignore the inactivated components, either by not being on the dashboard, or by not being in the details.  For RAI Responses, the package details should not have any details left over from the inactivated RAI Response.
7. Verify that re-activating the components rebuilds the package properly
